sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -277,10 +283,30 @@ function setName(entry: any, nameProperties: any): any {
 
     const parsedUrl = new URL(rawUrl);
     const pathname: string = parsedUrl.pathname;
-    const parts = pathname.split("/");
 
-    for (let i = 0; i * 2 < parts.length - 1; i++) {
-      entry[nameProperties[i]] = parts[i * 2 + 1];
+    const firstIndexOfDelimiter = pathname.indexOf("/");
+
+    if (pathname.match("(.*)/(.*)/Subscriptions/(.*)/Rules/(.*)")) {
```

Comment:
This is more complex than I thought it would be.

What I had in mind was splitting the string based on the `/` characted and then doing something like:
```
entry[nameProperties[0]] = parts[0]
entry[nameProperties[1]] = parts[2]
```

While the regex approach works, I think it is too complex to maintain and understand for what it needs to do.

Please consider simplifying.

Reply:
Simple parsing based on `/` would not work as `/` is allowed in the topic and queue names, so the features would fail for such requests.

Reply:
Ok.